src/com/google/enterprise/adaptor/DocIdPusher.java

```diff
@@ -215,8 +226,9 @@ public GroupPrincipal pushGroupDefinitions(
    * will wholly replace all existing definitions for the groupSource on the
    * GSA.
    *
-   * <p>A groupSource identifies the source of the group definitions. If
-   * provided, it must be a string of the form {@code [a-zA-Z_][a-zA-Z0-9_-]*}.
+   * <p>A groupSource identifies the source of the group definitions.
+   * If provided, it must be a string of the form
+   * <code>[a-zA-Z_][a-zA-Z0-9_-]{0,250}</code>.
```

Comment:
I think this is too technical, and you should stick to the * and give the max length separately.